gulpfile.js

```diff
@@ -2,15 +2,38 @@ const gulp = require('gulp');
 const babelify = require('babelify');
 const browserify = require('browserify');
 const source = require('vinyl-source-stream');
+const gutil = require('gulp-util');
+const notify = require('gulp-notify');
 
-gulp.task('default', ['css', 'browserify']);
+const reportError = function (error) {
+  notify({
+    title: 'Task Failed [browserify]',
+    message: 'See console.',
+    sound: 'Sosumi' // See: https://github.com/mikaelbr/node-notifier#all-notification-options-with-their-defaults
+  }).write(error);
+
+  gutil.beep();
+
+  // Pretty error reporting
+  let report = '';
+  const chalk = gutil.colors.white.bgRed;
+  const lineNumber = error.loc.line;
+
+  report += chalk('TASK:') + ' browserify\n';
+  report += chalk('PROB:') + ` ${error.message}\n`;
+  report += chalk('LINE:') + ` ${lineNumber}\n`;
```

Comment:
Instead of `error.message`, you might want to use `error.stack` here. It’ll generally be a lot more informative (and includes the message anyway). The stack will also include line number information, so you can probably drop the third line here.

Since the most common error will be a Babel `SyntaxError`, you might also want to special-case that, since it gives very nice output, but you don’t actually care about its stack trace:

```js
let message = error.stack;
if (error._babel && error instanceof SyntaxError) {
  // We don't care about Babel’s stack; we care about our bad code.
  // Remove the Babel stack at the bottom of the formatted code.
  message = error.stack.replace(/\n\s*at .*/g, '');
}

report += chalk('PROB:') + ` ${message}\n`;
```

Gets you:

<img width="906" alt="syntax-output" src="https://user-images.githubusercontent.com/74178/31651365-47b9dcf6-b2d0-11e7-8e19-3c45850efe1f.png">

Reply:
This is really awesome! Looks great and much more info. Thanks!